cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/TransformExecutorFactory.java

```diff
@@ -131,7 +138,15 @@ public TransformExecutorFactory(JobContext hadoopContext, PipelinePluginInstanti
       return;
     }
 
-    addTransformation(pipeline, stageName, transformations, transformErrorSinkMap);
+    try {
+      addTransformation(pipeline, stageName, transformations, transformErrorSinkMap);
+    } catch (Exception e) {
+      // Catch the Exception to generate a User Error Log for the Pipeline
+      PIPELINELOG.error("Failed to start Pipeline Stage '{}' with error: {}. Please review your Pipeline" +
+                          " configuration and check system logs for more details.", stageName,
+                        Throwables.getRootCause(e).getMessage(), Throwables.getRootCause(e));
```

Comment:
I'm proposing these styles:
- No use of initial capitalization unless "required" (such as "Hive" or "HBase")
- Always put the required space for joining two strings as the trailing space on the first string (rather than as the leading character of the second string)
```
+      PIPELINE_LOG.error("Failed to start pipeline stage '{}' with this error: {}. Please review your pipeline " +
+                          "configuration and check the system logs for details.", stageName,